### PR TITLE
[gettext] Add feature tools

### PR DESCRIPTION
**Pull request by Neumann-A**
required by some port to have gettext available. 
Be warned: Building tools requires a long time. 

supersedes #16612 @JackBoosY

### CODE REVIEW

**Comment by JackBoosY** [comment]
If you don't have enough time, I can handle this PR.

File: ports/gettext/portfile.cmake [hunk 1]
```diff
@@ -41,25 +42,42 @@ if(VCPKG_TARGET_IS_WINDOWS)
...
+                         DETERMINE_BUILD_TRIPLET
+                         USE_WRAPPERS
+                         ADD_BIN_TO_PATH    # So configure can check for working iconv
+                         OPTIONS --enable-relocatable #symbol duplication with glib-init.c?
+                                 --enable-c++
+                                 --disable-java
+                                 ${OPTIONS}
+                        )
+endif()
+
```
Comment (autoantwort) [inline ports/gettext/portfile.cmake:70]:
```suggestion
if("tools" IN_LIST FEATURES)
    set(ADDITIONAL_MSYS_PACKAGES gzip)
else()
    set(SOURCE_PATH ${SOURCE_PATH}/gettext-runtime)
endif()

vcpkg_configure_make(SOURCE_PATH ${SOURCE_PATH}
                     DETERMINE_BUILD_TRIPLET
                     USE_WRAPPERS
                     ADD_BIN_TO_PATH    # So configure can check for working iconv
                     OPTIONS --enable-relocatable #symbol duplication with glib-init.c?
                             --enable-c++
                             --disable-java
                             ${OPTIONS}
                     ADDITIONAL_MSYS_PACKAGES ${ADDITIONAL_MSYS_PACKAGES}
                    )

```
imho this is cleaner

**Comment by Neumann-A** [comment]
@JackBoosY Seems like CI is stuck for this PR

**Review by JackBoosY — COMMENTED** [review]
Note that only the following tools were generated:
```
autopoint       gettext         gettextize      msgcat          msgcomm         msgen           msgfilter       msggrep         msgmerge        msguniq         recode-sr-latin
envsubst        gettext.sh      msgattrib       msgcmp          msgconv         msgexec         msgfmt          msginit         msgunfmt        ngettext        xgettext
```

**Review by JackBoosY — APPROVED** [review]

**Comment by Neumann-A** [comment]
> Note that only the following tools were generated:

Only? isn't that everything? Autotools typically searches for `xgettext` and some of the `msg*` tools

**Comment by BillyONeal** [comment]
This regressed time to install gettext from 5 minutes to 45 minutes. :(

**Comment by BillyONeal** [comment]
Most expensive ports: 
image

**Comment by Neumann-A** [comment]
@BillyONeal  This didn't regress gettext. Before only gettext-runtime was build. With tools everything gettext has to offer is build. That is a significant difference. But i agree that there should be something done to make it faster. As far as I remember mist of the time is spend compressing some archive?

**Comment by BillyONeal** [comment]
>As far as I remember mist of the time is spend compressing some archive?

No idea. I do see a ton of .sh processes flying by though

**Pull request merged**